nixos/modules/services/computing/slurm/slurm.nix

```diff
@@ -0,0 +1,484 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.slurm;
+  # configuration file can be generated by http://slurm.schedmd.com/configurator.html
+  configFile = pkgs.writeText "slurm.conf" 
+    ''
+      ${optionalString (cfg.controlMachine != null) ''controlMachine=${cfg.controlMachine}''}
```

Comment:
Is all this needed? Just put the options that have a default as options, then let the user write the rest without options. E.g. mpiDefault and proctracktype have no default, then just don't create an option for it. Create options with defaults for a basic config to work.